server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.List;
+
+public enum TestFeatureVersion implements FeatureVersion {
+
+    TEST_0(0),
+    TEST_1(1),
+    TEST_2(2);
+
+    private short featureLevel;
+
+    public static final String FEATURE_NAME = "test.feature.version";
+    public static final TestFeatureVersion PRODUCTION_VERSION = TEST_1;
+
+    TestFeatureVersion(int featureLevel) {
+        this.featureLevel = (short) featureLevel;
+    }
```

Comment:
yes, assuming we specify metadata version as a numerical value and not a string as it is done in many places in the code.

as per the kip, validation may involve other features besides metadata version, so we will need to consider how that will be implemented.

(I didn't have any example of such interdependence, but it was requested that I support it)

Reply:
I have a plan for this but didn't quite get to it. Maybe we want to do this and maybe we don't. 

One idea is to have all features have a standard pattern/class for the fields. This would contain the metadata version mapping and a Map<String, Short> of the required other features. Not sure if we will need any more complicated validation logic than that. We could leave the opportunity to have a more complicated validate method if we choose. 

We can also show an example implementation in the TestFeatureVersion and folks can choose to replicate it.